tests/integration/test_cert_relation.py

```diff
@@ -178,13 +182,19 @@ async def test_given_additional_requirer_charm_deployed_when_relate_then_require
     )
     requirer_unit = model.units[f"{new_requirer_app_name}/0"]
 
-    action = await requirer_unit.run_action(action_name="get-certificate", hostname="any")
+    t0 = time.time()
+    timeout = 600
+    while time.time() - t0 < timeout:
```

Comment:
The while loop here is probably unnecessary because we are calling `wait_for_idle` beforehand.